src/wallet/wallet_rpc_server.cpp

```diff
@@ -1424,6 +1427,54 @@ namespace tools
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool wallet_rpc_server::on_start_mining(const wallet_rpc::COMMAND_RPC_START_MINING::request& req, wallet_rpc::COMMAND_RPC_START_MINING::response& res, epee::json_rpc::error& er)
+  {
+    if (!m_trusted_daemon)
+    {
+      er.code = WALLET_RPC_ERROR_CODE_UNKNOWN_ERROR;
+      er.message = "This command requires a trusted daemon.";
+      return false;
+    }
+
+    size_t max_mining_threads_count = (std::max)(tools::get_max_concurrency(), static_cast<unsigned>(2));
```

Comment:
This code is copied from [simplewallet.cpp L1608](https://github.com/monero-project/monero/blob/82411cdf3a35cef0518a6431bed016977f5deb50/src/simplewallet/simplewallet.cpp#L1608). Here, 2 is the minimum of threads count and the maximum is the result of `tools::get_max_concurrency()`.